R/lint.R

```diff
@@ -463,6 +481,23 @@ rstudio_source_markers <- function(lints) {
   out
 }
 
+check_ranges <- function(max_col, ranges) {
+  if (is.null(ranges)) {
+    return()
+  }
+  if (!is.list(ranges)) {
+    stop("`ranges` must be NULL or a list.")
+  } else if (!all(lengths(ranges) == 2L) || !all(vapply(ranges, is.numeric, logical(1L)))) {
```

Comment:
looks like we run 5-6 loops over `ranges`, how about

```r
if (!is.list(ranges)) {
  stop(...)
}
for (ii in seq_along(ranges)) check_range(ranges[[ii]])
```

and an individual range checker for each range that checks `length()`, `is.numeric()`, `anyNA()`, `is_valid_range()`?